### PR TITLE
#RI-6214 - migrate e2e to gh actions

### PR DESCRIPTION
**Pull request by zalenskiSofteq**
#RI-6214 - migrate e2e to gh actions

### CODE REVIEW

File: .github/workflows/tests-integration.yml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+        cp ./redisinsight/api/test/test-runs/coverage/test-run-result.json ./itest/results/${{ matrix.rte }}.result.json
+        cp ./redisinsight/api/test/test-runs/coverage/test-run-result.xml ./itest/results/${{ matrix.rte }}.result.xml
+        cp ./redisinsight/api/test/test-runs/coverage/test-run-coverage.json ./itest/coverages/${{ matrix.rte }}.coverage.json
+
+    # - name: Upload result files as artifact
+    #   if: always()
+    #   uses: actions/upload-artifact@v4
+    #   with:
+    #     name: results-${{ matrix.rte }}
+    #     path: itest/results
```
Comment (ArtemHoruzhenko) [inline .github/workflows/tests-integration.yml:124]: do we need this?

File: .github/workflows/tests-integration.yml [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+  run-tests:
+    name: ITest
+    runs-on: ubuntu-latest
+    needs: set-matrix
+    # container:
+    #   image: docker:latest
+    #   options: --privileged
+    #   volumes:
+    #     - /usr/src/app/coverage:/usr/src/app/coverage
```
Comment (ArtemHoruzhenko) [inline .github/workflows/tests-integration.yml:86]: do we need this?

File: .github/workflows/tests-integration.yml [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        npx nyc report -t ./coverages -r text -r text-summary
+        sudo mkdir -p /usr/src/app
+        sudo cp -a ./redisinsight/api/. /usr/src/app/
+        sudo cp -R ./coverages /usr/src/app && sudo chmod 777 -R /usr/src/app
+        cd /usr/src/app && npx nyc report -t ./coverages -r text -r text-summary
+
+    # - name: Setup upterm session
+    #   uses: mxschmitt/action-tmate@v3
+    #   with:
+    #     detached: true
```
Comment (ArtemHoruzhenko) [inline .github/workflows/tests-integration.yml:164]: and this

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -3,26 +3,37 @@ name: Tests
 on:
   push:
     branches:
-      - 'fe/*'
-      - 'be/*'
+      - 'fe/**'
+      - 'be/**'
+      - 'e2e/**'
     # branches-ignore:
     #   - main
```
Comment (ArtemHoruzhenko) [inline .github/workflows/tests.yml:10]: and this

File: .github/workflows/tests.yml [hunk 5]
```diff
@@ -48,18 +59,66 @@ jobs:
...
             - 'tests/e2e/**'
 
+  # TODO: concurrency
+
   frontend-tests:
-    # TODO: concurrency
-    # concurrency: build
     needs: changes
-    if: needs.changes.outputs.frontend == 'true' || inputs.all_tests || startsWith(github.ref_name, 'fe/')
+    # if: needs.changes.outputs.frontend == 'true'
```
Comment (ArtemHoruzhenko) [inline .github/workflows/tests.yml:66]: and this

File: tests/e2e/docker.web.docker-compose.yml [hunk 6]
```diff
@@ -7,8 +7,8 @@ services:
       dockerfile: e2e.Dockerfile
     tty: true
     volumes:
-      - ./results:/usr/src/app/results
-      - ./report:/usr/src/app/report
+      - /usr/src/app/results:/usr/src/app/results
+      - /usr/src/app/report:/usr/src/app/report
```
Comment (ArtemHoruzhenko) [inline tests/e2e/docker.web.docker-compose.yml:11]: we can't do this. it will be not possible to run tests locally

**Review by ArtemHoruzhenko — COMMENTED** [review]
please review and remove all comments

File: .github/workflows/tests-integration.yml (hunk 2 above)
Comment (zalenskiSofteq) [reply .github/workflows/tests-integration.yml]: removed

File: .github/workflows/tests-integration.yml (hunk 3 above)
Comment (zalenskiSofteq) [reply .github/workflows/tests-integration.yml]: removed

File: .github/workflows/tests.yml (hunk 4 above)
Comment (zalenskiSofteq) [reply .github/workflows/tests.yml]: removed

**Review by ArtemHoruzhenko — APPROVED** [review]

**Pull request merged**